Library/Homebrew/utils/curl.rb

```diff
@@ -303,6 +308,85 @@ def curl_http_content_headers_and_checksum(url, specs: {}, hash_needed: false, u
     def http_status_ok?(status)
       (100..299).cover?(status.to_i)
     end
+
+    # Separates the output text from `curl` into an array of response heads and
+    # the final response body.
+    # @param output [String] The output text from `curl` containing
+    #   response head(s), body, or both.
+    # @return [Hash] A hash containing an array of the response heads and the
+    #   body output, if found.
+    def parse_curl_output(output)
+      heads = []
+      return { heads: heads, body: "" } unless output.is_a?(String)
```

Comment:
@samford said:

> I'm not sure it would ever be anything other than a string but my original thinking was that this would allow us to gracefully handle bad input and avoid an error (since output needs to be a string for output.lstrip, output.match(...), and output.include?(...) to work).
>
> Looking at a later comment in the previous review, it does feel more appropriate to use a Sorbet type signature instead of this explicit guard, so I'll update this accordingly.

Reply:
Agreed, let's lean into Sorbet here instead of sniffing types. I also think this could probably `raise` rather than silently returning no headers/body.